optimade/models/baseinfo.py

```diff
@@ -48,17 +48,17 @@ class BaseInfoAttributes(BaseModel):
         "(i.e., the default is for is_index to be false).",
     )
 
-    @validator("entry_types_by_format", whole=True)
-    def formats_and_endpoints_must_be_valid(cls, v, values):
-        for format_, endpoints in v.items():
-            if format_ not in values["formats"]:
-                raise ValueError(f"'{format_}' must be listed in formats to be valid")
-            for endpoint in endpoints:
-                if endpoint not in values["available_endpoints"]:
-                    raise ValueError(
-                        f"'{endpoint}' must be listed in available_endpoints to be valid"
-                    )
-        return v
+    # @validator("entry_types_by_format", whole=True, check_fields=False)
+    # def formats_and_endpoints_must_be_valid(cls, v, values):
+    #     for format_, endpoints in v.items():
+    #         if format_ not in values["formats"]:
+    #             raise ValueError(f"'{format_}' must be listed in formats to be valid")
+    #         for endpoint in endpoints:
+    #             if endpoint not in values["available_endpoints"]:
+    #                 raise ValueError(
+    #                     f"'{endpoint}' must be listed in available_endpoints to be valid"
+    #                 )
+    #     return v
```

Comment:
The issue is class-inheritance.
This validator is run for sub-classes as well, as was intended, but it doesn't register and recognize the instance of the sub-class as that: An instance of the _sub_-class. Instead it thinks it's an instance of the parent class and fails.
In the `pydantic` docs [it's writen](https://pydantic-docs.helpmanual.io/usage/validators/#field-checks) that this can be avoided by passing the `check_fields=False` to the validator. But this is for `pydantic` v1, and didn't seem to work for me when I tested it here.
So I have commented it out, since it should be able to work for `pydantic` v1.

I don't know if this makes this PR blocked until we have upgraded or not, but it definitely not desireable to have a commented out model validator.